### DISCUSSION TITLE
How to pause and continue to run in sparta?

### DISCUSSION THREAD

**Discussion by CraneXJH** [sparcians/map#517] [Q&A] [answered]
Hi,
I use CommandLineSimulator to create a simulator. Then I want to implemnet pause function in the simulator. 
For Example, if I press 'p' in the keyboard, the simulator will pause and enter a sleep state. Then I press 'c' in the keyboard, it can continue to run.
Can the CommandLineSimulator acheive this function?

**Comment by klingaard** [comment]
The command line isn't designed to be interactive.  It's main use case is single testcase, single run with various modes to either run simulation completely or fractionally using command line options like `-i <inst cnt>` or `-r <cycle count>`.  

If you're looking for an interactive simulation, you can try rolling your own by bypassing the CommandLineSimulator instance and directly use the Simulation interface instead.   Also, there is python shell support you might try: https://github.com/sparcians/map/tree/master/sparta#python-shell-not-well-supported

**Reply by CraneXJH** [reply]
Thank you for your answer. 
I want to know if the sparta model can pause instead of stop? I found a function "stopRunning" in scheduler. I try to use this function, but the model will stop immediately. Is there a similar function can pause the scheduler?

**Comment by klingaard** [comment, marked as answer]
The scheduler allows you to call `stopRunning` at anytime and then you can call `run` immediately following that to continue simulation.  The Scheduler should continue where it left off, which can be in the middle of a time quantum or in between them.
